File: credit/credit.c
#include <stdio.h>
#include <cs50.h>

int main(){
    long num = 0;
    int checkSum = 1;

    do{
        num = get_long("What's your credit number? ");
    } while(num < 1);
    if(num >= 10000000000000000 || num < 1000000000000 || (num < 100000000000000 && num >= 10000000000000)){
        printf("INVALID\n");
    } else{
        // Calculate checkSum
        long counter = 10;
        long zwischen = 0;
        do{
           zwischen = num % counter;
           checkSum += zwischen / (counter / 10);
           counter *= 100;
           zwischen = 0;
        } while(num / (counter / 100) != 0);
        zwischen = 0;
        counter = 100;
        do{
           zwischen = num % counter;
           zwischen = (zwischen / (counter / 10)) * 2;
           if(zwischen < 10){
               checkSum += zwischen;
           } else {
               checkSum += zwischen / 10;
               checkSum += zwischen % 10;
           }
           counter *= 100;
           zwischen = 0;
        } while(num / (counter / 100) != 0);

        //checkSum abchecken
        if((checkSum - 1) % 10 != 0){
            printf("INVALID\n");
        }
        else{
            //get length and check if starts with right number
            //AMEX
            if(num / 10000000000000 == 34 || num / 10000000000000 == 37){
                printf("AMEX\n");
            } else
            //Mastercard
            if(num / 100000000000000 > 50 && num / 100000000000000 < 56){
                printf("MASTERCARD\n");
            } else
            //VISA
            if(num / 1000000000000000 == 4 || num / 1000000000000 == 4){
                printf("VISA\n");
            } else {
                printf("INVALID\n");
            }
        }
    }
}

File: PYTHON_credit/credit.py
from cs50 import get_string

num = get_string("Number: ")

if not (len(num) == 15 or len(num) == 16 or len(num) == 13):
    print("INVALID\n")
else:
    sum = 0
    for i in range (2, len(num) + 1, 2):
        zahl = 2 * int(num[-i])
        if zahl < 10:
            sum += zahl
        else:
            sum += zahl % 10
            sum += 1

    for i in range (1, len(num) + 1, 2):
        sum += int(num[-i])

    if not sum % 10 == 0:
        print("INVALID\n")
    else:
        if int(num[:2]) == 34 or int(num[:2]) == 37:
            print("AMEX\n")
        elif int(num[:2]) > 50 and int(num[:2]) < 56: #Mastercard
            print("MASTERCARD\n")
        elif int(num[:1]) == 4: #VISA
            print("VISA\n")
        else:
            print("INVALID\n")